### PR TITLE
Expose metrics container port for scraping

### LINKED ISSUE #2377: Expose ASOv2 metrics port(containerPort)
**Describe the current behavior**

Currently ASOv2 metrics are only available to be accessed from inside the pod. 

**Describe the improvement**

We want to expose the metrics port to make it available for other monitoring/tracing tools to use it.

### PR DESCRIPTION
**Pull request by super-harsh**
Closes #2377 

**What this PR does / why we need it**:

We did not have a way to expose metrics for scraping as metrics were only available internally to the pod. With this PR we have a service and ability to expose containerPort for metrics which enables the users to scrape the metrics.

**Special notes for your reviewer**:

- Includes fix for duplicate `.spec.template.metadata.annotations` field: Found out `.spec.template.metadata.annotations` field was being added twice as a part of script and recently added in the `config/manager/manager.yaml`.

- Changed the name for [auth_proxy_service](https://github.com/Azure/azure-service-operator/blob/main/v2/config/rbac/auth_proxy_service.yaml) to `azureserviceoperator-proxy-service`. As `azureserviceoperator-controller-manager-metrics-service` name did not quite make sense for a webhook proxy service.

### CODE REVIEW

**Review by theunrepentantgeek — APPROVED** [review]
Within the limits of my understanding about how metrics work, this looks reasonable. Needs a second approval from @matthchr before merging though, in case I missed something.

File: v2/config/default/manager_webhook_patch.yaml
Comment (matthchr) [reply v2/config/default/manager_webhook_patch.yaml:15]: I thought we did because we have `/healthz` on the `manager` container on port `8081 and that's the same container that runs the webhooks?

**Review by matthchr — APPROVED** [review]
LGTM other than the 1 minor question. Feel free to merge after either fixing that or confirming it's not actually a problem.

File: v2/config/default/manager_webhook_patch.yaml
Comment (super-harsh) [reply v2/config/default/manager_webhook_patch.yaml:15]: Having a generic healthz EP containerPort did not make sense to have it in this file as its for `webhook_patch`.. So rather moved it into `manager/manager.yaml`

**Pull request merged**
